attention_and_transformers.ipynb

```diff
@@ -0,0 +1,4395 @@
+{
```

Comment:
<p>[Suggestion] tip if they're familiar with einsum is to use it.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deep-learning-indaba/indaba-pracs-2022/pull/5/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='34'/>

Reply:
<p>I love einsum, but I feel here it shows very specifically the steps required, instead of hiding it in the einsum setup. But damn was hard not to just use it everywhere haha. </p>